PMSCore/Tester.cs

```diff
@@ -0,0 +1,77 @@
+﻿namespace PMSCore
+{
+    /// <summary>
+    /// A Tester user who manages test cases and bug reporting.
+    /// </summary>
+    public class Tester : UserBase
+    {
+        // List to store test case descriptions
+        public List<string> TestCases { get; private set; }
+
+        // List to store bug reports with description and severity
+        public List<(string Description, string Severity)> BugReports { get; private set; }
+
+        /// <summary>
+        /// Initializes a Tester user with a username and TESTER permission level.
+        /// </summary>
+        public Tester(string userName) : base(userName, Permission.TESTER)
+        {
+            TestCases = new List<string>();
+            BugReports = new List<(string Description, string Severity)>();
+        }
+
+        /// <summary>
+        /// Adds a new test case if the description is valid.
+        /// </summary>
+        /// <param name="testCaseDescription">Description of the test case.</param>
+        public void AddTestCase(string testCaseDescription)
+        {
+            if (!string.IsNullOrWhiteSpace(testCaseDescription))
+            {
+                TestCases.Add(testCaseDescription);
+                // Replace Console.WriteLine with logging for better testability
+                _logger.LogInformation($"Test case added: {testCaseDescription}");
+            }
+            else
+            {
+                _logger.LogWarning("Test case description should not be empty.");
+            }
+        }
+
+        /// <summary>
+        /// Returns the list of test cases added by the tester.
+        /// </summary>
+        /// <returns>List of test case descriptions.</returns>
+        public List<string> GetTestCases()
+        {
+            return new List<string>(TestCases);
```

Comment:
Even if this was to keep this would create a new list everytime and copy the TestCases list into we would occupy double the necessary memory, return TestCases would suffice I believe